Quests/BaseQuest.cs

```diff
@@ -64,28 +73,61 @@ private void GiveRewards(Player player)
             }
         }
 
+        /// <summary>
+        /// Resets the kill count of the NPC within the Dictionary after completion
+        /// </summary>
+        /// <param name="player"></param>
+        private void ResetKillCount(Player player)
+        {
+            var modPlayer = player.GetModPlayer<QuestPlayer>();
+            var KilledList = modPlayer.KilledNPCs;
+            foreach (IQuestRequirement requirement in Requirements)
+            {
+                if (requirement is OrRequirement orRequirement)
+                {
+                    foreach (KillRequirement kill in orRequirement.clauses)
+                    {
+                        KilledList[kill.type] = 0;
```

Comment:
A neater way to handle this would be to have a `ResetState(Player)` method on IQuestRequirement, but we can look at that later. I have some thoughts for how to generalize quest state completely.